src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,385 @@
+      this.emitRatingChange();
+      this.determineActiveItems();
+    }
+  }
+
+  private determineInitialRating() {
+    const valueToUse = this.value ? this.value : this.average;
+    this.ratingItems?.forEach((item) => {
+      item.dataset.hovered = "false";
+      item.dataset.average =
+        this.average && !this.value && parseInt(item.dataset.value) <= this.average
+          ? "true"
+          : "false";
+      item.dataset.selected =
+        item.dataset.average === "false" && parseInt(item.dataset.value) <= valueToUse
+          ? "true"
+          : "false";
+      item.icon =
+        parseInt(item.dataset.value) <= valueToUse || this.readOnly
+          ? this.selectedIconType
+          : this.iconType;
+    });
+
+    if (this.average) {
+      this.determinePartialStar();
+    }
+  }
+
+  private determinePartialStar() {
+    // position an extra icon and fill to width of percentage
+    const rootVal = Math.floor(this.average);
+    const decimal = parseInt((this.average % 1).toFixed(2).split(".")[1]);
+
+    // track the closest full star to position the partial icon
+    this.ratingItems?.forEach((item) => {
+      if (this.average && !this.value && parseInt(item.dataset.value) === rootVal + 1) {
+        this.rootStar = item;
+        this.rootStar.dataset.partialparent = "true";
+      }
+    });
+
+    if (this.rootStar) {
+      this.partialStar.dataset.rootvalue = (rootVal + 1).toString();
+
+      // get margin from first star item in case partial is last and has none
+      const marginProp = this.dir === "rtl" ? "margin-left" : "margin-right";
+      const margin = window.getComputedStyle(this.ratingItems[0]).getPropertyValue(marginProp);
+      const offset = `calc((${this.rootStar.offsetWidth}px + ${margin}) * ${rootVal})`;
+
+      this.partialStarContainer.style.width = `${(decimal / 100) * this.rootStar.offsetWidth}px`;
+      this.partialStarContainer.style.height = `${this.rootStar.offsetHeight}px`;
+
+      this.dir === "rtl"
+        ? (this.partialStarContainer.style.right = offset)
+        : (this.partialStarContainer.style.left = offset);
+    }
+  }
+
+  private resetHoverState() {
+    if (this.partialStarContainer) this.partialStarContainer.dataset.partialhidden = "false";
+    this.ratingItems?.forEach((item) => {
+      item.dataset.hovered = "false";
+    });
+  }
+
+  private determineActiveItems() {
+    const valueToUse = this.value ? this.value : this.average;
+    this.ratingItems?.forEach((item) => {
+      item.dataset.hovered = "false";
+      item.dataset.average = "false";
+      item.dataset.selected = parseInt(item.dataset.value) <= valueToUse ? "true" : "false";
+      item.icon =
+        this.readOnly || parseInt(item.dataset.value) <= this.value
+          ? this.selectedIconType
+          : this.iconType;
+    });
+  }
+
+  private showSelectedIconOnHover(e) {
+    if (!this.readOnly) {
+      // handle edge and ie not interpreting mapped stars
+      const isEdgeOrIE =
```

Comment:
nitpick: Seems like this could be a method since it's used twice.

Reply:
Done.